### ISSUE TITLE
light color theme

### ISSUE THREAD

**Issue by nbiederbeck** [arcticicestudio/nord#46]
Is there an option to make a light color theme based on nord which just switches background/foreground color?
I'm thinking of something like [solarized](https://github.com/altercation/solarized).

**Comment by arcticicestudio** [comment]
I planned to provide a kind of "daylight" mode/variant for all port projects, but this is still in the distant future. There are almost 10 or more issues/emails per port project I maintain each day and it costs a lot of time and effort to keep them up-to-date.

The only change for Nord itself would be the style guide to specify/document the color behavior in case of light mode/layout.

I've added a [backlog issue card](https://github.com/arcticicestudio/nord/projects/1#card-3697219) and will notify you as soon as I start working on this.

**Comment by lokesh-krishna** [comment]
> There are almost 10 or more issues/emails per port project I maintain and it costs a lot of time and effort to keep them up-to-date  

Could you consider updating the style guide here and opening issues on all the port projects so that contributors can follow the guide here and create a daylight variant for those projects? You would obviously review them before adding them to the respective port project so that the quality of the project as a whole isn't affected.

**Comment by arcticicestudio** [comment]
Sorry for the late reply.

I need to create a concept about the usage of the colors first in order to plan the implementation for all the port projects. I think it won't take long, but I'm currently focusing on getting all the open issues and port project requests done before working on the next "large" feature, where "large" means that it will take a lot of time to realize it for all port projects.

**Comment by bemug** [comment]
I'm really interested in this.
Is it still on track?

**Comment by arcticicestudio** [comment]
@bemug This is still planned, but there are still several other issues in the [backlog][bl] and [next version][nv] boards. I'm also busy with many real life tasks (job, study etc.) and unfortunately there's not much free time to work on my projects. When there's some time on the weekend or vacation I focus to get the [Nord docs and website][docs] done (if there are no other critical/high priority issues in any project).

[bl]: https://github.com/arcticicestudio/nord/projects/1
[nv]: https://github.com/arcticicestudio/nord/projects/4
[docs]: https://github.com/arcticicestudio/nord-docs/issues/1

**Comment by lancelet** [comment]
Someone has already ventured a `nord-light` theme for `doom-emacs`:
  - https://www.reddit.com/r/emacs/comments/8a0joj/share_my_first_light_theme_doomnordlighttheme/
  - https://github.com/hlissner/emacs-doom-themes/blob/master/themes/doom-nord-light-theme.el

(I'm adding it to the comments here because I really love both the light and dark Nord themes in doom and was hoping to find iTerm2 equivalents.)

**Comment by bemug** [comment]
Bumping this issue. Is it still planned?

**Comment by arcticicestudio** [comment]
@bemug It is still planned, but there are many more urgent and essential tasks that need to be done before starting to work on a new and larger feature like this. The plan is to migrate docs and assets of all existing port projects to the [new shiny Nord website](https://www.nordtheme.com) to finally have a single-point-of-truth™ for documentations and assets for Nord. Afterwards important and long-time outstanding issues and PRs in all port projects need to be reviewed/implemented, merged and deployed in new release versions. When everything is done, new and large features like a light theme variant/mode are the next step to go. After that, the many requests for new port projects will be processed that'll come with the "dark" theme mode as well as the new light variant by default.

**Comment by ghost** [comment]
There is another theme, [polar](https://github.com/mtyn/polar), which attempts to do something like this. It could be a nice starting point for an official light theme.

**Comment by mvrogowski** [comment]
Any updates on this?

**Comment by arcticicestudio** [comment]
@mvrogowski You can track the progress of the data transition in https://github.com/arcticicestudio/nord-docs/issues/133.

**Comment by ghost** [comment]
Will light theme lead to changes in the Nord color palette? The syntax highlighting colors look bad on the Snowstorm palette.

**Comment by arcticicestudio** [comment]
The colors won't change. Nord was designed with a darker appearance in mind in the first place so it might be possible that some colors are not optimal, but creating minimal palettes that work perfectly in universal modes is almost impossible to achieve. Some port projects might make use of color calculation functions (e.g. Sass `darken()`) to improve some elements that use colors that are not matching fine, but it'll need evaluation and tests when starting to work on light theme variants.

Nord makes use of the awesome [Refactoring UI][rui] design pattern. Here are some great resources about dark and light mode theme styles:

> When implementing dark mode, don’t throw away the visual cues in the light version by naively inverting the color scheme.
[![](https://pbs.twimg.com/media/D_m97tRW4AADwti?format=png&name=4096x4096)][r1]

> I simply use darker shades for dark themes and lighter shades for light themes.
[![](https://pbs.twimg.com/media/D0BpDWEW0AE8YnD?format=jpg&name=medium)][r2]

[r1]: https://twitter.com/steveschoger/status/1151160261170126850?lang=en
[r2]: https://twitter.com/steveschoger/status/1098995277124517890?lang=en
[rui]: https://refactoringui.com

**Comment by bemug** [comment]
Bumping this as we're now in July and I'm still interested in this to happen.
I see this issue is still in the backlog. Did it progress?

**Comment by brorbw** [comment]
Shameless self plug. I made one since I couldn't find one. Its based on the Nord light theme for doom emacs and the dark theme for iterm
[https://github.com/brorbw/nord-light-iterm2](https://github.com/brorbw/nord-light-iterm2)

**Comment by arcticicestudio** [comment]
@brorbw Thanks for sharing, always nice to see the community comes up with own projects 😄 

@bemug In short: No. In long:
At the time of writing this there are currently [over 220 open issues][i] and over [50 open PRs][pr] (also note the ~1200+ closed issues and ~600 PRs) just in the repositories of this account (`/arcticicestudio`). Next to “normal“ issues and PRs the backlog includes requests for [over 50 port projects for various applications and tools][ppr]. Each repository also has it's own “Backlog“ project board which includes notes and reminders for open tasks and future work. In addition [each repository also has a “Support“ that contains issue with many different purposes][s] not related to the project/code itself like questions regarding the usage, installation process or requests for help with problems. Other communication channels which provide a way for users to ask for help, e.g. [the official Nord communities][com] like [Reddit][r], [Keybase][kb] or even direct requests to my email, are another source of time consumers that are not directly “visible to the outside“.
  
All of that together costs time and often requires more steps like bug reproduction or testing before merging changes from PRs (automatic tests are kind of hard when it comes to projects that are mostly “visual“ and might behave different on various platforms in combination with different environments and setups).
I know that the same and appearance of this account might indicate that this could be a kind of organization or company, but in fact [it's “just“ me][sp] with a “legacy“ history why this account exists as it is today and I still work on a larger post to clarify all of this. If you add it all up it means I'd need to drop out of my current full time job (spoiler: won't do) and leave some of the “real life“ activities behind in order to get keep up with this workload. Also I won't add more details about the fact that there are other projects next to Nord where I contribute to which also requires time.

In summary this means that the implementation of “light“ variants are still further down in the list of prioritized tasks for Nord while there are still plenty of other ones like the [data migration of all existing port projects to the new website and documentation infrastructure][ppdt], “cost optimizations“ for such to not go beyond the limit of 100GB traffic per month (currently at ~80GB) in the free _Netlify_ hosting plan (at least not as long as no one sponsors it), the [implementation of the “brighter comment color“][bc] for remaining port projects, overall code cleanup and unification to reduce the maintenance overhead even more, the implementation of requested port project, the…
The list goes on, and until these are not completed another large tasks like touching all port projects to implement a "visually correct“ _light mode_ is not justifiable.

I hope you can understand my situation, but comments like “why not implemented yet?“ or “is it done?“ that add another entry to my daily 20-40+ GitHub notifications can be kind of nerve-wracking. Don't get me wrong here please, the issue is still planned and not rejected in any way, but when working on larger projects one learns that prioritization is key to not end up in chaos.
Wow, wall-of-text comment again 😄 

[i]: https://github.com/issues?q=is%3Aopen+is%3Aissue+archived%3Afalse+user%3Aarcticicestudio
[pr]: https://github.com/issues?q=is%3Aopen+is%3Apr+archived%3Afalse+user%3Aarcticicestudio
[ppr]: https://github.com/arcticicestudio/nord/projects/1#column-288113
[s]: https://github.com/pulls?q=archived%3Afalse+user%3Aarcticicestudio+label%3Atype-support
[sp]: https://github.com/sponsors/arcticicestudio
[com]: https://www.nordtheme.com/community
[r]: https://www.reddit.com/r/nordtheme
[kb]: https://keybase.io/team/nord.chat
[ppdt]: https://github.com/arcticicestudio/nord-docs/issues/133
[bc]: https://github.com/arcticicestudio/nord/issues/94

**Comment by bemug** [comment]
Thanks a lot for your comment. Keep up the good work!

**Comment by brorbw** [comment]
@arcticicestudio I don't mean to plagiarize the theme. I think I have come pretty close to the one in [doom emacs nord light](https://www.reddit.com/r/emacs/comments/8a0joj/share_my_first_light_theme_doomnordlighttheme/). If you are up for it, i could make a PR in [Nord Iterm](https://github.com/arcticicestudio/nord-iterm2). 

Since maintaining different ports of the same theme is cumbersome and time consuming while working fulltime might I suggest getting the community to help out by inviting others to help maintain it. Another way would be to automate the ports. You could have the colors defined in one config file, and build all the ports based on that file. This way you could invite maintainers to work on the build process and not on specific projects. This would also mean you have a single source of truth for all the ports. I can imagine having different projects you could end up having a fragmented theme that's not consistent over multiple platforms.

**Comment by kvkeerthi** [comment]
it will be nice to get Nord Light Theme for Pycharm Editor

**Comment by brorbw** [comment]
> it will be nice to get Nord Light Theme for Pycharm Editor

@kvkeerthi iirc Pycharm is a JetBrains product - https://www.nordtheme.com/ports/jetbrains

Edit: Just realised that there is no light theme for JetBrains products

**Comment by atahrijouti** [comment]
Hi @arcticicestudio, is there anyway we can help to make the light version come to life? Or go up in priorities?

**Comment by arcticicestudio** [comment]
@abderrahmane-tj The amount of open tasks is really high like the [400+ open issues and PRs][1], community platform management and long-time planned maintenance-overhead improvements. The focus on a large topic this completely new theme variant requires a lot of time and has to be planned for all port projects. The best way to help are contributions to open tasks since 90% of my free time I can spend on Nord it taken by support/help topics and the fact that new features and improvements are requested, but never contributed by anyone else.

[1]: https://github.com/issues?q=is%3Aopen+user%3Aarcticicestudio+sort%3Aupdated-desc

**Comment by lastzero** [comment]
We took the liberty of creating a light theme based on the Nord color palette for [PhotoPrism](https://www.photoprism.app/). It's called "Snow" and you can try it on [our public demo](https://demo.photoprism.app/library/settings):

![Screenshot 2023-02-19 at 11 29 15](https://user-images.githubusercontent.com/301686/219942684-fa62941a-b2c7-4952-97f3-b9351d2efed2.png)

**Comment by bakert** [comment]
I took a stab at a Nord light theme for Terminal.app on MacOS – https://github.com/bakert/nord-light

![image](https://github.com/user-attachments/assets/89fe9a6a-9363-4c76-9dbc-029374011458)

Any comments gratefully received :)
